src/listen.ts

```diff
@@ -99,8 +106,18 @@ export async function listen(
   } else {
     server = createServer(handle);
     addShutdown(server);
-    // @ts-ignore
-    await promisify(server.listen.bind(server))(port, listhenOptions.hostname);
+
+    listhenOptions.ipc
+      ? await promisify(server.listen.bind(server))(
+          // @ts-ignore
+          ipcSocket,
+        )
+      : await promisify(server.listen.bind(server))(
+          // @ts-ignore
+          port,
+          listhenOptions.hostname,
```

Comment:
Refactor: Since only options are different, we might construct listhenOptions object conditionally and pass it here.

Reply:
It's possible to reduce the function calls to 1 with rest operator and with an iterator per object. It's still a bit messy:

```js
const args = typeof listhenOptions.ipc === 'string'
      ? { ipcSocket, *[Symbol.iterator]() { yield this.ipcSocket } }
      : { port, hostname: listhenOptions.hostname, *[Symbol.iterator]() { yield this.port; yield this.hostname } }

    // @ts-ignore
    await promisify(server.listen.bind(server))(...args)
```

Should I check this in?

Reply:
I just checked it in.. Like you can see :grimacing:

Reply:
Just checked it in